compute/instances/creating-instance-with-custom-machine-type/create_instance_with_custom_shared_core_with_helper.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_custom_machine_type_create_shared_with_helper]
+import (
+	"fmt"
+	"io"
+)
+
+// createInstanceWithCustomSharedCore creates a new VM instance with a custom type using shared CPUs.
+func createInstanceWithCustomSharedCore(w io.Writer, projectID, zone, instanceName, cpuSeries string, memory int, tl TypeLimit) error {
```

Comment:
this example is not self-contained, as it requires `createCustomMachineType` and `createInstanceWithCustomMachineType`.
It is also not clear to me what you are demonstrating with this snippet, so i am unable to offer more concrete suggestions to fix.